### PR TITLE
(update) Owners dynamically updates occupancy and drinks

### PR DESCRIPTION
**Pull request by lschwall**
Customer List in owner's view renders drink totals and they can add a drink to that total. The totals will reflect the updated count upon adding one drink

In Owner's profile, owners can now see at a glance their occupency level

### CODE REVIEW

**Review by Bri-B — APPROVED** [review]

File: server/routes/customer.js [hunk 1]
```diff
@@ -187,13 +187,12 @@ customerRouter.get('/findMe', (req, res) => {
 
 customerRouter.get('/getFriendById', (req, res) => {
   const { customerId } = req.query;
-  console.info('Grabbing information for customerID: ', customerId)
+  // console.info('Grabbing information for customerID: ', customerId)
```
Comment (jstrzesz) [inline server/routes/customer.js]: take this out

File: server/routes/drinks.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    const { id, barId } = req.body;
+    Drinks.create({
+        id_customer: id,
+        drink_Count: 1,
+        id_bar: barId
+    })
+})
+
+drinksRouter.put('/updateCount', (req, res) => {
+    // const { count, id, barId } = req.body;
```
Comment (jstrzesz) [inline server/routes/drinks.js]: remove, it's the same line as the one below it.

File: server/routes/drinks.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    const { count, id, barId } = req.body
+    // console.log(count, id, barId)
```
Comment (jstrzesz) [inline server/routes/drinks.js]: remove

File: server/routes/drinks.js [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    Drinks.increment({ drink_Count: 1 },
+        {
+            where: {
+                id_customer: id,
+                id_bar: barId
+            }
+        })
+        .then((drinktotal) => {
```
Comment (jstrzesz) [inline server/routes/drinks.js:40]: there needs to be a .catch block after the .then

File: server/routes/drinks.js [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+            res.send(JSON.stringify(drinktotal[0][0][0].drink_Count));
+        })
+})
+
+drinksRouter.get('/count', (req, res) => {
+    const { customerId, barId } = req.query;
+    Drinks.findOne({ where: { id_customer: customerId, id_bar: barId } })
+        .then((customerDrinks) => {
```
Comment (jstrzesz) [inline server/routes/drinks.js:51]: again you need a .catch black

File: src/components/Owner/AddDrink.jsx [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+                        .then(({ data }) => setDrinkCount(data))
+                        .catch(err => console.log(err))
+                }
+            })
+            .catch((err) => console.warn(err))
+    }
+
+    const drinkTotal = () => {
+        axios.get(`/db/drinks/count?customerId=${id}&barId=${barId}`)
+            .then(({ data }) => {
```
Comment (jstrzesz) [inline src/components/Owner/AddDrink.jsx:26]: need a .catch block

File: src/components/Owner/AddDrink.jsx [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+    }
+
+    useEffect(() => {
+        drinkTotal();
+    }, [])
+
+
+    const updateCount = () => {
+        setCount(count + 1);
+        console.log(count)
```
Comment (jstrzesz) [inline src/components/Owner/AddDrink.jsx]: remove console.log

File: src/components/Owner/AddDrink.jsx [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+    }
+
+    return (
+        <div>
+            {}
```
Comment (jstrzesz) [inline src/components/Owner/AddDrink.jsx]: why are there empty curly braces here?

File: src/components/Owner/CustomerEntry.jsx [hunk 9]
```diff
@@ -1,15 +1,15 @@
...
-const CustomerEntry = ({ customer }) => {
+const CustomerEntry = ({ customer, barId }) => {
 
     const [customerData, setCustomerData] = useState();
+    const [drink, setDrinkCount] = useState('');
     const getCustomerData = () => {
-        console.log(typeof customer, 'CUSTOMER')
         axios.get(`/db/customer/getFriendById?customerId=${customer}`)
             .then(({ data }) => {
                 console.log(data, 'DATA')
```
Comment (jstrzesz) [inline src/components/Owner/CustomerEntry.jsx]: take out console.log

File: src/components/Owner/OwnerProfile.jsx [hunk 10]
```diff
@@ -50,31 +56,33 @@ const OwnerProfile = ({ setViewValue, barId, count }) => {
...
-                setBarAddress(data[0].address)
-                setBarNumber(data[0].phone_number)
-                setOcc(data[0].capacity);
-            })
+    let covidCap = (capacity * .25)
+    let covidCapLow = covidCap * .25;
+    let covidCapHigh = covidCap * .75
+    const occupencyStatus = () => {
+        if (count < covidCapLow) {
+            console.log(count, (covidCap * .25), 'COUNT')
```
Comment (jstrzesz) [inline src/components/Owner/OwnerProfile.jsx]: take out console.log. Your linter/husky should not be allowing this push to go through with all of these console.logs in

File: src/components/Owner/OwnerView.jsx [hunk 11]
```diff
@@ -29,20 +38,69 @@ const OwnerView = ({ setViewValue, barId }) => {
...
 
     const handleChange = (event, newValue) => {
         setValue(newValue);
     };
 
+    const idArr = [];
+
+    const barInfo = () => {
+        axios.get(`/db/bar/info?id=${barId}`)
+            .then(({ data }) => {
```
Comment (jstrzesz) [inline src/components/Owner/OwnerView.jsx:58]: need a .catch block after .then

File: src/components/Owner/OwnerView.jsx [hunk 12]
```diff
@@ -29,20 +38,69 @@ const OwnerView = ({ setViewValue, barId }) => {
...
+                setBarName(data[0].bar_name);
+                setBarAddress(data[0].address)
+                setBarNumber(data[0].phone_number)
+                setCapacity(data[0].bar_capacity);
+            })
+    }
+
+    const fetchCustomers = () => {
+        axios.get(`/db/cb/list?barId=${barId}`)
+            .then(({ data }) => {
```
Comment (jstrzesz) [inline src/components/Owner/OwnerView.jsx:72]: need a .catch block after the then

File: src/components/Owner/BarCustomerList.jsx [hunk 13]
```diff
@@ -1,32 +1,18 @@
...
-        }, 2000)
-    }, [])
+import CustomerEntry from './CustomerEntry.jsx';
+import PhoneIcon from '@material-ui/icons/Phone';
+import PlusOneIcon from '@material-ui/icons/PlusOne';
 
+const BarList = ({ customerList, barId }) => {
     if (customerList.length === 0) {
         return (
             <Grid container direction="column" justify="center" alignItems="center">
```
Comment (jstrzesz) [inline src/components/Owner/BarCustomerList.jsx:8]: move all of the rendered logic to a separate component and then pass down props where needed

File: src/components/Owner/CustomerEntry.jsx [hunk 14]
```diff
@@ -31,11 +31,14 @@ const CustomerEntry = ({ customer }) => {
                             {`${customerData.first_name} ${customerData.last_name}`}
                         </Typography>
                     </Grid>
-                    <Grid item container direction="column" justify="center" alignItems="center">
+                    <Grid item container direction="column" justify="center" alignItems="center" xs={3}>
```
Comment (jstrzesz) [inline src/components/Owner/CustomerEntry.jsx:33]: all rendered jsx should be moved to separate component and instantiated here with props passed down

File: src/components/Owner/OwnerProfile.jsx [hunk 15]
```diff
@@ -50,31 +56,33 @@ const OwnerProfile = ({ setViewValue, barId, count }) => {
...
+            )
+        }
     }
 
-    useEffect(() => {
-        barInfo();
-    }, [])
-
     return (
         <Grid container direction="column" justify="center" alignItems="center">
```
Comment (jstrzesz) [inline src/components/Owner/OwnerProfile.jsx:84]: see other comments about rendered jsx

**Review by jstrzesz — CHANGES_REQUESTED** [review]
see all comments

**Review by tenhold — APPROVED** [review]

File: src/components/Owner/AlertEntry.jsx [hunk 16]
```diff
@@ -0,0 +1,22 @@
+import React, { useState, useEffect } from 'react';
+import axios from 'axios';
+
+const AlertEntry = ({ customer, barId, count }) => {
+
+    const [state, setState] = useState();
+
+    useEffect(() => {
+        axios.get(`/db/drinks/alerts?customer=${customer}&barId=${barId}&count=${count}`)
+            .then(({ data }) => {
```
Comment (jstrzesz) [inline src/components/Owner/AlertEntry.jsx:10]: another missing catch block

**Review by christophermbooth — APPROVED** [review]

**Pull request merged**